plugins/dell-dock/fu-dell-dock-i2c-ec.c

```diff
@@ -204,14 +227,19 @@ gboolean
 fu_dell_dock_ec_needs_tbt (FuDevice *device)
 {
 	FuDellDockEc *self = FU_DELL_DOCK_EC (device);
-	gboolean port0_tbt_mode = self->data->port0_dock_status & TBT_MODE_MASK;
+	gboolean port0_tbt_mode = self->data->port0_dock_status &
+		( DOCK_MODE_CAP_TBT | DOCK_MODE_CAP_USB4 );
 
 	/* check for TBT module type */
-	if (self->data->module_type != MODULE_TYPE_130_TBT &&
-	    self->data->module_type != MODULE_TYPE_45_TBT)
-		return FALSE;
+	switch (self->data->module_type){
+		case MODULE_TYPE_130_TBT:
+		case MODULE_TYPE_45_TBT:
+		case MODULE_TYPE_130_USB4:
```

Comment:
I'm not convinced this case should be here - isn't this code going to cause an I2C endpoint to be made?  You don't want that for the USB4 module.

Reply:
That's right, I don't need i2c device for usb4 regardless what the port status is.

Reply:
Done.